### PR TITLE
chore(docs): update contribution guide

### PR DESCRIPTION
**Pull request by icfantv**
* update contribution guide to include section on how to propose and discuss
  new features.

### CODE REVIEW

**Comment by icfantv** [comment]
Hey guys, please feel free to change the wording as you deem appropriate.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -43,3 +43,10 @@ We are always looking for the quality contributions and will be happy to accept
...
   * your PR doesn't break the build; check the Travis-CI build status after opening a PR and push corrective commits if anything goes wrong
   * your commits conform to the conventions established [here](https://github.com/ajoslin/conventional-changelog/blob/master/conventions/angular.md)
+* If you have an idea for, or the code you want to contribute is, a new feature, or changes the way an existing feature works (possibly a breaking change), please open an issue to start the discussion so that we may collaborate on an acceptable solution if one exists.
+  * we would prefer you not submit a PR with the new feature or change without first discussing with us - we value your time and this is to ensure you aren't wasting it.
+  * you should be prepared to fully justify your proposal - this means it should be well thought out.
+  * even if all of the above are followed, there is no guarantee we will be able to implement or accept your proposal or solution
+    * please understand that if we do this it's not because we are on a power trip or want to be mean
```
Comment (Foxandxss) [inline CONTRIBUTING.md]: missing period.

Comment (icfantv) [inline CONTRIBUTING.md]: doh!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -43,3 +43,10 @@ We are always looking for the quality contributions and will be happy to accept
   * your directive has accompanying tests and all the tests are passing; don't hesitate to contact us (angular-ui@googlegroups.com) if you need any help with unit testing
   * your PR doesn't break the build; check the Travis-CI build status after opening a PR and push corrective commits if anything goes wrong
   * your commits conform to the conventions established [here](https://github.com/ajoslin/conventional-changelog/blob/master/conventions/angular.md)
+* If you have an idea for, or the code you want to contribute is, a new feature, or changes the way an existing feature works (possibly a breaking change), please open an issue to start the discussion so that we may collaborate on an acceptable solution if one exists.
+  * we would prefer you not submit a PR with the new feature or change without first discussing with us - we value your time and this is to ensure you aren't wasting it.
```
Comment (wesleycho) [inline CONTRIBUTING.md]: Is this indentation difference intentional?

Comment (Foxandxss) [inline CONTRIBUTING.md]: Asked myself the same.

Comment (wesleycho) [inline CONTRIBUTING.md]: Also I think there needs to be a caveat here - I'm perfectly fine if someone opens a PR if it is very well thought out or correcting a small error (i.e. typo in the docs).

Comment (icfantv) [inline CONTRIBUTING.md]: Yes.  I'm totally open to changes if you feel it doesn't make sense or is weird.

Comment (icfantv) [inline CONTRIBUTING.md]: oh.  i guess the others have `:` and the bullets (at least for the middle one) are a sentence continuation...

Comment (icfantv) [inline CONTRIBUTING.md]: @wesleycho, agreed.  did you have some text in mind?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -43,3 +43,10 @@ We are always looking for the quality contributions and will be happy to accept
   * your directive has accompanying tests and all the tests are passing; don't hesitate to contact us (angular-ui@googlegroups.com) if you need any help with unit testing
   * your PR doesn't break the build; check the Travis-CI build status after opening a PR and push corrective commits if anything goes wrong
   * your commits conform to the conventions established [here](https://github.com/ajoslin/conventional-changelog/blob/master/conventions/angular.md)
+* If you have an idea for, or the code you want to contribute is, a new feature, or changes the way an existing feature works (possibly a breaking change), please open an issue to start the discussion so that we may collaborate on an acceptable solution if one exists.
+  * unless it's well thought out or correcting a small error (e.g. a typo in the docs) or we would prefer you not submit a PR with the new feature or change without first discussing with us - we value your time and this is to ensure you aren't wasting it.
```
Comment (wesleycho) [inline CONTRIBUTING.md]: I would avoid contractions such as `it's` and `aren't` here, but perhaps I'm being a little pedantic - avoiding contractions is viewed as more professional.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -43,3 +43,10 @@ We are always looking for the quality contributions and will be happy to accept
...
+  * you should be prepared to fully justify your proposal - this means it should be well thought out.
```
Comment (wesleycho) [inline CONTRIBUTING.md]: I would leave out the part after the `-` - IMO that is already duplicated in the prior bullet & implied here.

**Comment by icfantv** [comment]
@wesleycho, updated again.  i like the idea of letting this bake for a while so we get the wording right.  @pkozlowski-opensource, i know you're busy, but if you have any thoughts, they would be welcome.  cheers.

**Comment by wesleycho** [comment]
I'm fine with this, although I can't say I'm the greatest fan of the tone it sets. What are your thoughts @Foxandxss ?

**Comment by icfantv** [comment]
Should we just forego it then?  I certainly don't want to convey an air of "we don't need you" but on the flipside it's helpful to be able to point to a policy for clarification.

**Comment by Foxandxss** [comment]
I kinda agree with Wesley.

I think that there is (or should be) a general understanding in how a FOSS project work. People should know already that they cannot create a big PR and assume it will get merged.

So I think we are just re-describing what a FOSS project normally does.

**Comment by icfantv** [comment]
Ok.  Closing then.  Thanks guys.

**Pull request closed without merging**